runtime/src/main/java/org/corfudb/protocols/logprotocol/MultiObjectSMREntry.java

```diff
@@ -93,39 +103,115 @@ public void mergeInto(MultiObjectSMREntry other) {
+                        .publishPercentileHistogram(true)
+                        .register(registry));
         int startIdx = b.writerIndex();
-        super.serialize(b);
-        b.writeInt(streamUpdates.size());
-        streamUpdates.entrySet().stream()
-                .forEach(x -> {
-                    //int streamStart = b.writerIndex();
-                    b.writeLong(x.getKey().getMostSignificantBits());
-                    b.writeLong(x.getKey().getLeastSignificantBits());
-                    Serializers.CORFU.serialize(x.getValue(), b);
-                });
-
+        try {
+            super.serialize(b);
+            b.writeInt(streamUpdates.size());
+            streamUpdates.entrySet().stream()
+                    .forEach(x -> {
+                        int streamStart = b.writerIndex();
+                        Optional<Timer> serializeStreamTimer = MeterRegistryProvider
+                                .getInstance().map(registry -> Timer.builder(METRIC_PREFIX + "." + "serialize.stream")
+                                        .tags(STREAM_ID, x.getKey().toString())
+                                        .publishPercentiles(TRACKED_PERCENTILES)
+                                        .publishPercentileHistogram(true)
+                                        .register(registry));
+                        Optional<Timer.Sample> serializeStreamSample =
+                                MeterRegistryProvider.getInstance().map(Timer::start);
+                        Optional<DistributionSummary> streamDistributionSummary = MeterRegistryProvider.getInstance().map(registry ->
+                                DistributionSummary.builder(METRIC_PREFIX + "." + "serialize.stream.size")
+                                        .tags(STREAM_ID, x.getKey().toString())
+                                        .publishPercentiles(TRACKED_PERCENTILES)
+                                        .publishPercentileHistogram(true)
+                                        .register(registry));
+                        try {
+                            b.writeLong(x.getKey().getMostSignificantBits());
+                            b.writeLong(x.getKey().getLeastSignificantBits());
+                            Serializers.CORFU.serialize(x.getValue(), b);
+                        } finally {
+                            serializeStreamSample.ifPresent(sample -> serializeStreamTimer.ifPresent(sample::stop));
+                            streamDistributionSummary.ifPresent(summary -> summary.record(b.writerIndex() - streamStart));
+                        }
+                    });
+        } finally {
+            serializeSample.ifPresent(sample -> serializeTimer.ifPresent(sample::stop));
+            distributionSummary.ifPresent(summary -> summary.record(b.writerIndex() - startIdx));
+            distributionSummaryEntries.ifPresent(summary -> summary.record(streamUpdates.size()));
```

Comment:
This will record the number of streams in a MultiOBjectSMR, you probably also want to record the number of updates within a stream, that is, ` x.getValue().getUpdates().size()`